# Ultralytics Framework for YOLOv10 Training

In [1]:
import torch

# Number of GPUs available
num_gpus = torch.cuda.device_count()

print(f"Number of available GPUs: {num_gpus}")


Number of available GPUs: 4


# Yolov10n Model Training on:
- Carla_Labelling-v5 dataset
- Train: 70% = 4485 Images
- Validation: 20% = 1274 Images
- Test: 10% = 641 Images

- Hyperparameters are on automatic settings using Ultralytics package

In [2]:
from ultralytics import YOLO

# Load YOLOv10n model from scratch
model = YOLO("yolov10n.pt")

100%|██████████| 5.59M/5.59M [00:00<00:00, 160MB/s]


# Validate the pretrained weights by running inference on an image.

In [3]:
# Run batched inference on a list of images
results = model(["/home/sur06423/project/1.jpg"])  # return a list of Results objects

# Process results list
for result in results:
    boxes = result.boxes  # Boxes object for bounding box outputs
    masks = result.masks  # Masks object for segmentation masks outputs
    keypoints = result.keypoints  # Keypoints object for pose outputs
    probs = result.probs  # Probs object for classification outputs
    obb = result.obb  # Oriented boxes object for OBB outputs
    result.show()  # display to screen
    result.save(filename="result.jpg")  # save to disk


0: 448x640 4 horses, 63.4ms
Speed: 6.0ms preprocess, 63.4ms inference, 26.6ms postprocess per image at shape (1, 3, 448, 640)


Error: no "view" rule for type "image/png" passed its test case
       (for more information, add "--debug=1" on the command line)


# Train the model using the following guidelines:
- Link for various arguments: https://docs.ultralytics.com/modes/train/#train-settings 

In [4]:
import os

# Set library paths
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg-nvidia-525.116.04/lib/x86_64-linux-gnu:' + os.environ.get('LD_LIBRARY_PATH', '')
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg/lib/x86_64-linux-gnu:' + os.environ['LD_LIBRARY_PATH']
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg-nvidia-535.113.01/lib/x86_64-linux-gnu:' + os.environ['LD_LIBRARY_PATH']

# Verify the update
print(os.environ['LD_LIBRARY_PATH'])


/usr/lib/xorg-nvidia-535.113.01/lib/x86_64-linux-gnu:/usr/lib/xorg/lib/x86_64-linux-gnu:/usr/lib/xorg-nvidia-525.116.04/lib/x86_64-linux-gnu:/home/sur06423/miniconda3/envs/yolov_env/lib/python3.9/site-packages/cv2/../../lib64:


In [5]:
# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
model.train(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", # 	Path to the dataset configuration file (e.g., coco8.yaml). This file contains dataset-specific parameters, including paths to training and validation data, class names, and number of classes.
            epochs=100, # Total number of training epochs. Each epoch represents a full pass over the entire dataset. Adjusting this value can affect training duration and model performance.
            patience=10, # Number of epochs to wait without improvement in validation metrics before early stopping the training. Helps prevent overfitting by stopping training when performance plateaus.
            batch=128, # Batch size, with three modes: set as an integer (e.g., batch=16), auto mode for 60% GPU memory utilization (batch=-1), or auto mode with specified utilization fraction (batch=0.70).
            imgsz=640, # Target image size for training. All images are resized to this dimension before being fed into the model. Affects model accuracy and computational complexity.
            save=True, # Enables saving of training checkpoints and final model weights. Useful for resuming training or model deployment.
            save_period = 1, # Frequency of saving model checkpoints, specified in epochs. A value of -1 disables this feature. Useful for saving interim models during long training sessions.
            device=[0,1,2,3], # Specifies the computational device(s) for training: a single GPU (device=0), multiple GPUs (device=0,1), CPU (device=cpu), or MPS for Apple silicon (device=mps).
            workers=6, # Number of worker threads for data loading (per RANK if Multi-GPU training). Influences the speed of data preprocessing and feeding into the model, especially useful in multi-GPU setups.
            project="/home/sur06423/project/Version_5_exp/YOLO_v10/v10_n", # Name of the project directory where training outputs are saved. Allows for organized storage of different experiments.
            pretrained = True, # Determines whether to start training from a pretrained model. Can be a boolean value or a string path to a specific model from which to load weights. Enhances training efficiency and model performance.
            optimizer = "auto", # Choice of optimizer for training. Options include SGD, Adam, AdamW, NAdam, RAdam, RMSProp etc., or auto for automatic selection based on model configuration. Affects convergence speed and stability.
            verbose = True, # Enables verbose output during training, providing detailed logs and progress updates. Useful for debugging and closely monitoring the training process.
            seed = 0, # Sets the random seed for training, ensuring reproducibility of results across runs with the same configurations.
            amp = True, # Enables Automatic Mixed Precision (AMP) training, reducing memory usage and possibly speeding up training with minimal impact on accuracy.
            profile =True, # Enables profiling of ONNX and TensorRT speeds during training, useful for optimizing model deployment.
            val=True, # Enables validation during training, allowing for periodic evaluation of model performance on a separate dataset.
            plots = True, #Generates and saves plots of training and validation metrics, as well as prediction examples, providing visual insights into model performance and learning progression.                
)

New https://pypi.org/project/ultralytics/8.3.13 available 😃 Update with 'pip install -U ultralytics'
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:3 (NVIDIA RTX A6000, 48677MiB)


engine/trainer: task=detect, mode=train, model=yolov10n.pt, data=/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml, epochs=100, time=None, patience=10, batch=128, imgsz=640, save=True, save_period=1, cache=False, device=[0, 1, 2, 3], workers=6, project=/home/sur06423/project/Version_5_exp/YOLO_v10/v10_n, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=True, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, 

100%|██████████| 6.25M/6.25M [00:00<00:00, 124MB/s]


AMP: checks passed ✅


train: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/train/labels.cache... 4485 images, 430 backgrounds, 0 corrupt: 100%|██████████| 4485/4485 [00:00<?, ?it/s]
val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/valid/labels.cache... 1274 images, 124 backgrounds, 0 corrupt: 100%|██████████| 1274/1274 [00:00<?, ?it/s]


Plotting labels to /home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.000714, momentum=0.9) with parameter groups 95 weight(decay=0.0), 108 weight(decay=0.001), 107 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 24 dataloader workers
Logging results to /home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      1/100      6.28G       2.86      8.177      2.127          5        640: 100%|██████████| 36/36 [00:11<00:00,  3.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.23it/s]


                   all       1274       7377    0.00723      0.121     0.0613     0.0529

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      2/100      6.17G      2.579      4.835       1.98         17        640: 100%|██████████| 36/36 [00:09<00:00,  3.76it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.92it/s]


                   all       1274       7377      0.929     0.0535      0.117     0.0896

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      3/100      6.02G      2.531      4.111       1.99         37        640: 100%|██████████| 36/36 [00:09<00:00,  3.79it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.89it/s]


                   all       1274       7377      0.941      0.087      0.133     0.0973

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      4/100      6.07G      2.474      3.563      1.989          8        640: 100%|██████████| 36/36 [00:09<00:00,  3.76it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.76it/s]


                   all       1274       7377      0.738      0.178      0.266      0.157

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      5/100      6.27G      2.496      3.031      1.964          6        640: 100%|██████████| 36/36 [00:09<00:00,  3.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.85it/s]


                   all       1274       7377      0.532      0.362      0.364       0.21

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      6/100      6.06G      2.457      2.575      1.939         26        640: 100%|██████████| 36/36 [00:09<00:00,  3.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.67it/s]


                   all       1274       7377      0.557      0.391        0.4      0.222

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      7/100      6.19G      2.408      2.353       1.93          3        640: 100%|██████████| 36/36 [00:09<00:00,  3.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.93it/s]


                   all       1274       7377      0.606      0.385      0.405      0.227

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      8/100      6.09G      2.434      2.168      1.911         24        640: 100%|██████████| 36/36 [00:09<00:00,  3.71it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.58it/s]


                   all       1274       7377      0.626      0.396      0.435      0.257

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      9/100      6.11G      2.315      2.009       1.92          1        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.82it/s]


                   all       1274       7377      0.637      0.418      0.457      0.263

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     10/100      6.03G      2.368      1.913      1.895          8        640: 100%|██████████| 36/36 [00:09<00:00,  3.96it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.94it/s]


                   all       1274       7377      0.639      0.422      0.469      0.279

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     11/100      6.09G      2.398      1.867      1.881         14        640: 100%|██████████| 36/36 [00:09<00:00,  3.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.95it/s]


                   all       1274       7377      0.637      0.406      0.446      0.263

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     12/100      6.08G      2.359       2.03      1.859          1        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.90it/s]


                   all       1274       7377       0.67       0.42      0.467      0.282

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     13/100      6.17G      2.302      1.728      1.881         11        640: 100%|██████████| 36/36 [00:09<00:00,  3.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.77it/s]


                   all       1274       7377      0.664      0.464      0.499        0.3

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     14/100      6.02G      2.334      1.752      1.867         14        640: 100%|██████████| 36/36 [00:09<00:00,  3.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.92it/s]


                   all       1274       7377      0.683      0.464       0.51      0.307

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     15/100      6.09G      2.286      1.646      1.862         24        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.95it/s]


                   all       1274       7377      0.681      0.447      0.499      0.285

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     16/100      6.15G      2.283      1.662       1.87          4        640: 100%|██████████| 36/36 [00:09<00:00,  3.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.01it/s]


                   all       1274       7377      0.693      0.453      0.503       0.31

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     17/100      6.16G      2.257      1.603      1.876         15        640: 100%|██████████| 36/36 [00:09<00:00,  3.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.98it/s]


                   all       1274       7377      0.748      0.437      0.519      0.312

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     18/100      6.16G      2.222      1.576      1.882          6        640: 100%|██████████| 36/36 [00:09<00:00,  4.00it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.92it/s]


                   all       1274       7377      0.692      0.441       0.49      0.294

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     19/100      6.39G      2.283      1.579      1.862         12        640: 100%|██████████| 36/36 [00:09<00:00,  3.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.97it/s]


                   all       1274       7377      0.725      0.479      0.534      0.324

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     20/100      6.36G      2.213      1.531      1.853         23        640: 100%|██████████| 36/36 [00:09<00:00,  3.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.92it/s]


                   all       1274       7377      0.699      0.466      0.516      0.307

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     21/100      6.16G      2.174      1.495      1.842          4        640: 100%|██████████| 36/36 [00:09<00:00,  3.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.93it/s]


                   all       1274       7377      0.711      0.449      0.515      0.322

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     22/100      6.33G      2.285      1.545      1.847          3        640: 100%|██████████| 36/36 [00:09<00:00,  3.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.89it/s]


                   all       1274       7377       0.73      0.484      0.537       0.33

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     23/100      5.93G      2.175      1.465      1.837         27        640: 100%|██████████| 36/36 [00:09<00:00,  3.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.93it/s]


                   all       1274       7377      0.681      0.499      0.545      0.326

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     24/100      6.08G      2.141      1.436      1.835          3        640: 100%|██████████| 36/36 [00:08<00:00,  4.03it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.94it/s]


                   all       1274       7377      0.695      0.492      0.535      0.323

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     25/100       6.1G      2.138       1.43      1.831         20        640: 100%|██████████| 36/36 [00:09<00:00,  3.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.89it/s]


                   all       1274       7377      0.758      0.504      0.566      0.337

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     26/100      5.88G      2.112      1.414      1.846         18        640: 100%|██████████| 36/36 [00:09<00:00,  3.92it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.93it/s]


                   all       1274       7377      0.717      0.503      0.554      0.342

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     27/100      6.15G        2.2      1.396      1.829         11        640: 100%|██████████| 36/36 [00:08<00:00,  4.01it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.94it/s]


                   all       1274       7377      0.703      0.502      0.559      0.339

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     28/100      6.24G      2.122      1.379      1.831         20        640: 100%|██████████| 36/36 [00:09<00:00,  3.96it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.87it/s]


                   all       1274       7377       0.72      0.506      0.569      0.351

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     29/100      6.33G      2.075      1.325      1.809          7        640: 100%|██████████| 36/36 [00:09<00:00,  3.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.93it/s]


                   all       1274       7377      0.736      0.514      0.571      0.351

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     30/100      6.07G      2.123      1.352      1.832         11        640: 100%|██████████| 36/36 [00:09<00:00,  3.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.91it/s]


                   all       1274       7377      0.729      0.514      0.575      0.353

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     31/100      6.31G      2.071      1.338      1.811         19        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.98it/s]


                   all       1274       7377       0.77      0.508      0.576      0.356

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     32/100      6.32G      2.111      1.321      1.804         10        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.02it/s]


                   all       1274       7377      0.749      0.518      0.583       0.36

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     33/100      6.46G      2.061      1.328      1.811         13        640: 100%|██████████| 36/36 [00:09<00:00,  3.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.05it/s]


                   all       1274       7377      0.733        0.5      0.566      0.354

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     34/100      6.13G      2.085      1.298      1.823         11        640: 100%|██████████| 36/36 [00:09<00:00,  3.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.01it/s]


                   all       1274       7377      0.754      0.521      0.579      0.353

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     35/100      6.37G      2.092      1.311      1.828         21        640: 100%|██████████| 36/36 [00:09<00:00,  3.76it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.93it/s]


                   all       1274       7377      0.745      0.535      0.589      0.362

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     36/100      6.32G      2.069      1.299      1.811          9        640: 100%|██████████| 36/36 [00:09<00:00,  3.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.95it/s]


                   all       1274       7377      0.764      0.526      0.587       0.36

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     37/100      6.02G      2.064      1.299      1.825         10        640: 100%|██████████| 36/36 [00:09<00:00,  3.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.01it/s]


                   all       1274       7377      0.747      0.528      0.596      0.369

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     38/100      6.05G      1.964      1.285      1.768          0        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.99it/s]


                   all       1274       7377       0.77      0.529      0.599      0.374

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     39/100      6.29G      2.045      1.255      1.802         21        640: 100%|██████████| 36/36 [00:09<00:00,  3.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.94it/s]


                   all       1274       7377      0.762      0.543      0.608      0.375

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     40/100      6.26G      2.024      1.266      1.798         15        640: 100%|██████████| 36/36 [00:09<00:00,  3.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.85it/s]


                   all       1274       7377       0.74      0.539      0.598       0.37

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     41/100         6G       2.02      1.269      1.814         10        640: 100%|██████████| 36/36 [00:09<00:00,  3.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.88it/s]


                   all       1274       7377      0.782      0.526      0.601      0.374

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     42/100      6.12G      2.026      1.249      1.802         24        640: 100%|██████████| 36/36 [00:09<00:00,  3.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.95it/s]


                   all       1274       7377      0.761      0.537      0.602      0.378

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     43/100      6.13G      1.985      1.247      1.816          4        640: 100%|██████████| 36/36 [00:09<00:00,  3.96it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.00it/s]


                   all       1274       7377      0.788      0.547      0.615      0.384

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     44/100      6.55G      2.035      1.224      1.776         26        640: 100%|██████████| 36/36 [00:09<00:00,  3.79it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.85it/s]


                   all       1274       7377      0.785      0.524        0.6      0.375

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     45/100      6.13G      1.962      1.205      1.791          8        640: 100%|██████████| 36/36 [00:09<00:00,  3.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.99it/s]


                   all       1274       7377       0.76      0.536      0.603      0.379

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     46/100      6.33G      1.931      1.165      1.784          8        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.92it/s]


                   all       1274       7377      0.748      0.545      0.608      0.386

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     47/100         6G      1.954      1.191      1.782         15        640: 100%|██████████| 36/36 [00:09<00:00,  3.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.95it/s]


                   all       1274       7377      0.776      0.532      0.603      0.383

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     48/100      6.06G          2      1.178       1.78         16        640: 100%|██████████| 36/36 [00:09<00:00,  3.92it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.96it/s]


                   all       1274       7377      0.779      0.548      0.614       0.38

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     49/100      6.16G      1.988      1.338      1.758          7        640: 100%|██████████| 36/36 [00:09<00:00,  3.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.90it/s]


                   all       1274       7377      0.771      0.546      0.609      0.379

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     50/100      6.05G      2.047      1.197      1.787         17        640: 100%|██████████| 36/36 [00:09<00:00,  3.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.93it/s]


                   all       1274       7377       0.79      0.543      0.617      0.392

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     51/100      6.16G      1.978      1.159      1.795         10        640: 100%|██████████| 36/36 [00:09<00:00,  3.69it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.91it/s]


                   all       1274       7377      0.802      0.544      0.619      0.391

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     52/100      6.12G      1.933       1.14      1.791          5        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.02it/s]


                   all       1274       7377        0.8      0.547      0.627      0.396

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     53/100      5.97G      1.926      1.167      1.784          8        640: 100%|██████████| 36/36 [00:09<00:00,  3.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.94it/s]


                   all       1274       7377      0.772      0.562      0.626      0.393

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     54/100      6.03G      1.878       1.12      1.766         14        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.81it/s]


                   all       1274       7377      0.797      0.551      0.624      0.396

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     55/100      6.14G      1.946      1.178      1.778         21        640: 100%|██████████| 36/36 [00:09<00:00,  3.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.93it/s]


                   all       1274       7377      0.788      0.565      0.627      0.401

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     56/100      6.28G       1.88      1.118      1.781          5        640: 100%|██████████| 36/36 [00:09<00:00,  3.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.01it/s]


                   all       1274       7377      0.779       0.55      0.627      0.401

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     57/100      6.17G      1.923       1.12      1.762         13        640: 100%|██████████| 36/36 [00:08<00:00,  4.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.03it/s]


                   all       1274       7377      0.779      0.573      0.632      0.397

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     58/100       6.1G      1.895      1.117      1.778         21        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.93it/s]


                   all       1274       7377      0.795      0.557      0.634      0.398

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     59/100      6.06G      1.915      1.101      1.766         12        640: 100%|██████████| 36/36 [00:09<00:00,  3.96it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.97it/s]


                   all       1274       7377      0.773      0.567      0.628      0.401

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     60/100      6.21G      1.915      1.141      1.768         18        640: 100%|██████████| 36/36 [00:09<00:00,  3.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.01it/s]


                   all       1274       7377      0.772      0.575      0.635      0.407

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     61/100      6.07G      1.937      1.137      1.769         39        640: 100%|██████████| 36/36 [00:09<00:00,  3.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.99it/s]


                   all       1274       7377      0.784      0.543      0.619      0.394

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     62/100      6.28G      1.859      1.106       1.77         33        640: 100%|██████████| 36/36 [00:09<00:00,  3.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.94it/s]


                   all       1274       7377      0.794      0.563      0.634      0.407

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     63/100      6.07G      1.908      1.133      1.765         13        640: 100%|██████████| 36/36 [00:09<00:00,  3.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.96it/s]


                   all       1274       7377      0.796      0.558      0.627      0.406

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     64/100       6.1G      1.942      1.113      1.756          6        640: 100%|██████████| 36/36 [00:09<00:00,  3.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.91it/s]


                   all       1274       7377      0.769       0.57      0.634      0.402

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     65/100      6.13G      1.918      1.119      1.769          3        640: 100%|██████████| 36/36 [00:09<00:00,  3.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.00it/s]


                   all       1274       7377      0.791      0.574      0.642      0.408

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     66/100      6.26G      1.897      1.119      1.773         10        640: 100%|██████████| 36/36 [00:09<00:00,  3.83it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.91it/s]


                   all       1274       7377      0.793      0.578      0.641       0.41

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     67/100      6.35G      1.937       1.13      1.769         19        640: 100%|██████████| 36/36 [00:09<00:00,  3.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.97it/s]


                   all       1274       7377      0.803      0.572      0.636      0.401

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     68/100       6.1G       1.88      1.063      1.746         19        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.94it/s]


                   all       1274       7377      0.798      0.582      0.643      0.409

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     69/100      6.09G      1.908        1.1      1.769         16        640: 100%|██████████| 36/36 [00:09<00:00,  3.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.95it/s]


                   all       1274       7377      0.812      0.568      0.644      0.408

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     70/100      6.01G       1.88      1.067      1.762         24        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.79it/s]


                   all       1274       7377        0.8      0.573      0.644      0.411

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     71/100      6.34G      1.867      1.065      1.764         17        640: 100%|██████████| 36/36 [00:09<00:00,  3.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.88it/s]


                   all       1274       7377      0.796      0.574      0.641      0.414

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     72/100      6.29G      1.854      1.058      1.755         12        640: 100%|██████████| 36/36 [00:09<00:00,  3.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  5.00it/s]


                   all       1274       7377      0.805      0.571      0.643      0.415

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     73/100      6.26G      1.883      1.062      1.751         13        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.89it/s]


                   all       1274       7377       0.81      0.579      0.644      0.417

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     74/100      6.03G      1.802      1.041      1.757         11        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.85it/s]


                   all       1274       7377      0.803       0.58      0.641      0.414

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     75/100      6.33G      1.853      1.056      1.753          6        640: 100%|██████████| 36/36 [00:09<00:00,  3.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.88it/s]


                   all       1274       7377      0.803      0.581      0.649      0.418

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     76/100      6.13G       1.88      1.059      1.751          8        640: 100%|██████████| 36/36 [00:09<00:00,  3.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.87it/s]


                   all       1274       7377      0.786      0.589      0.655       0.42

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     77/100      6.13G      1.863       1.07       1.75         22        640: 100%|██████████| 36/36 [00:09<00:00,  3.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.05it/s]


                   all       1274       7377      0.789      0.582      0.649      0.422

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     78/100      6.31G      1.831      1.011      1.773          4        640: 100%|██████████| 36/36 [00:09<00:00,  3.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.04it/s]


                   all       1274       7377      0.807      0.586      0.653      0.422

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     79/100      6.25G      1.828       1.03      1.758         13        640: 100%|██████████| 36/36 [00:09<00:00,  3.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.04it/s]


                   all       1274       7377      0.793      0.582      0.647      0.418

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     80/100      6.33G        1.8      1.043      1.746         36        640: 100%|██████████| 36/36 [00:09<00:00,  3.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.05it/s]


                   all       1274       7377      0.788      0.587      0.651       0.42

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     81/100       6.5G      1.809      1.028      1.749          7        640: 100%|██████████| 36/36 [00:09<00:00,  3.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.01it/s]


                   all       1274       7377      0.809      0.575      0.648      0.418

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     82/100      6.24G       1.81      1.031      1.753          9        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.98it/s]


                   all       1274       7377      0.776      0.593       0.65       0.42

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     83/100      6.35G      1.826      1.019      1.749          8        640: 100%|██████████| 36/36 [00:09<00:00,  3.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.88it/s]


                   all       1274       7377      0.786      0.586      0.655      0.423

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     84/100      6.07G       1.83      1.023      1.754          8        640: 100%|██████████| 36/36 [00:09<00:00,  3.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.91it/s]


                   all       1274       7377       0.81      0.586      0.656      0.423

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     85/100      6.13G      1.841      1.023      1.759         11        640: 100%|██████████| 36/36 [00:09<00:00,  3.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.97it/s]


                   all       1274       7377      0.832      0.573      0.651      0.421

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     86/100         6G      1.877      1.113      1.871          3        640: 100%|██████████| 36/36 [00:09<00:00,  3.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.87it/s]


                   all       1274       7377      0.809       0.58      0.653      0.425

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     87/100      5.97G      1.819      1.003      1.753         14        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.92it/s]


                   all       1274       7377      0.795      0.585      0.653      0.425

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     88/100      6.13G      1.822     0.9982      1.745         16        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.81it/s]


                   all       1274       7377      0.822      0.577      0.655      0.425

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     89/100      6.28G      1.711      0.945      1.747         14        640: 100%|██████████| 36/36 [00:09<00:00,  3.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  5.00it/s]


                   all       1274       7377      0.811      0.582      0.657      0.429

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     90/100      6.14G      1.758      1.012      1.765          3        640: 100%|██████████| 36/36 [00:09<00:00,  3.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.99it/s]


                   all       1274       7377       0.82      0.582      0.656      0.426
Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     91/100      6.12G      1.894      1.001      1.725         13        640: 100%|██████████| 36/36 [00:10<00:00,  3.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.03it/s]


                   all       1274       7377      0.814      0.585      0.655      0.424

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     92/100       6.1G      1.892      1.006      1.724         13        640: 100%|██████████| 36/36 [00:09<00:00,  3.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.02it/s]


                   all       1274       7377      0.811      0.588      0.658      0.427

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     93/100         6G      1.884     0.9872      1.718          7        640: 100%|██████████| 36/36 [00:09<00:00,  3.84it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.02it/s]


                   all       1274       7377      0.819      0.588       0.66      0.429

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     94/100      6.14G      1.888      1.009      1.721          6        640: 100%|██████████| 36/36 [00:08<00:00,  4.05it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.02it/s]


                   all       1274       7377      0.814      0.589       0.66       0.43

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     95/100      6.18G      1.933      1.025      1.713         31        640: 100%|██████████| 36/36 [00:09<00:00,  3.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.89it/s]


                   all       1274       7377      0.806      0.591      0.659      0.429

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     96/100      6.01G      1.879      1.021      1.722         13        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.99it/s]


                   all       1274       7377        0.8       0.59      0.658       0.43

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     97/100      6.01G      1.891     0.9812      1.715         13        640: 100%|██████████| 36/36 [00:09<00:00,  3.92it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.98it/s]


                   all       1274       7377      0.821      0.587      0.661       0.43

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     98/100      6.01G      1.858       0.97        1.7          3        640: 100%|██████████| 36/36 [00:08<00:00,  4.01it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.03it/s]


                   all       1274       7377      0.832      0.583      0.661      0.433

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     99/100         6G      1.861     0.9657      1.707         40        640: 100%|██████████| 36/36 [00:09<00:00,  3.96it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.98it/s]


                   all       1274       7377      0.821      0.586      0.661      0.433

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    100/100      6.09G      1.873     0.9891      1.717          6        640: 100%|██████████| 36/36 [00:09<00:00,  3.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.93it/s]


                   all       1274       7377       0.81      0.591      0.661      0.432

100 epochs completed in 0.421 hours.
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train/weights/last.pt, 5.8MB
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train/weights/best.pt, 5.8MB

Validating /home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train/weights/best.pt...
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:3 (NVIDIA RTX A6000, 48677MiB)
YOLOv10n summary (fused): 285 layers, 2,695,976 parameters, 0 gradients, 8.2 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:07<00:00,  2.73it/s]


                   all       1274       7377      0.829      0.583      0.661      0.433
            Pedestrian        333        600      0.752      0.582      0.653      0.371
         Traffic_Signs        309        434      0.803      0.495      0.566      0.335
               Vehicle       1034       2820      0.901      0.839      0.901      0.713
         traffic_light        660       3523       0.86      0.418      0.525      0.315
Speed: 0.1ms preprocess, 1.2ms inference, 0.0ms loss, 1.3ms postprocess per image
Results saved to /home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train


# Results on Validation set:
100 epochs completed in 0.421 hours.
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train/weights/last.pt, 5.8MB
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train/weights/best.pt, 5.8MB

Validating /home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train/weights/best.pt...
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:3 (NVIDIA RTX A6000, 48677MiB)
YOLOv10n summary (fused): 285 layers, 2,695,976 parameters, 0 gradients, 8.2 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:07<00:00,  2.73it/s]
                   all       1274       7377      0.829      0.583      0.661      0.433
            Pedestrian        333        600      0.752      0.582      0.653      0.371
         Traffic_Signs        309        434      0.803      0.495      0.566      0.335
               Vehicle       1034       2820      0.901      0.839      0.901      0.713
         traffic_light        660       3523       0.86      0.418      0.525      0.315
Speed: 0.1ms preprocess, 1.2ms inference, 0.0ms loss, 1.3ms postprocess per image
Results saved to /home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train

# Test the results on Test dataset
- The above results was on the validation set of the Carla_labeling-v5 dataset (See the Image counts)
- Now we need to calculate the metrics on the test set of the Carla_labeling-v5 dataset containing only 641 images.
- Below is the script to do so:

In [6]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train/weights/last.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLOv10n summary (fused): 285 layers, 2,695,976 parameters, 0 gradients, 8.2 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:09<00:00,  1.55s/it]


                   all        641       3701      0.804      0.572      0.722      0.517
            Pedestrian        182        332      0.707      0.566      0.687      0.445
         Traffic_Signs        168        230      0.809      0.478      0.672      0.446
               Vehicle        528       1377      0.887      0.831      0.899      0.769
         traffic_light        338       1762      0.812      0.411      0.631       0.41
Speed: 0.4ms preprocess, 2.2ms inference, 0.0ms loss, 0.1ms postprocess per image
Saving runs/detect/val/predictions.json...
Results saved to runs/detect/val


# Results on Last checpoint (100th epoch):
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLOv10n summary (fused): 285 layers, 2,695,976 parameters, 0 gradients, 8.2 GFLOPs
val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:09<00:00,  1.55s/it]
                   all        641       3701      0.804      0.572      0.722      0.517
            Pedestrian        182        332      0.707      0.566      0.687      0.445
         Traffic_Signs        168        230      0.809      0.478      0.672      0.446
               Vehicle        528       1377      0.887      0.831      0.899      0.769
         traffic_light        338       1762      0.812      0.411      0.631       0.41
Speed: 0.4ms preprocess, 2.2ms inference, 0.0ms loss, 0.1ms postprocess per image
Saving runs/detect/val/predictions.json...
Results saved to runs/detect/val

# Results on Test Set using Best Checkpoint:

In [7]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load the model
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)


YOLOv10n summary (fused): 285 layers, 2,695,976 parameters, 0 gradients, 8.2 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:09<00:00,  1.58s/it]


                   all        641       3701      0.809      0.562      0.719      0.516
            Pedestrian        182        332      0.713      0.557      0.681      0.443
         Traffic_Signs        168        230      0.811      0.465      0.665      0.445
               Vehicle        528       1377      0.895      0.825      0.899      0.768
         traffic_light        338       1762      0.816        0.4      0.629      0.409
Speed: 0.4ms preprocess, 2.6ms inference, 0.0ms loss, 0.1ms postprocess per image
Saving runs/detect/val2/predictions.json...
Results saved to runs/detect/val2


In [15]:
print(checkpoint.keys())

dict_keys(['date', 'version', 'license', 'docs', 'epoch', 'best_fitness', 'model', 'ema', 'updates', 'optimizer', 'train_args', 'train_metrics', 'train_results'])


# Taking the Optimal value from the F1-Confidence curve for Optimal Confidence
- Optimal Confidence YOLOv10n - 0.264
- IoU threshold : 0.5
- Results save d at: runs/detect/val3

In [1]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train/weights/last.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.264, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.5, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLOv10n summary (fused): 285 layers, 2,695,976 parameters, 0 gradients, 8.2 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:07<00:00,  1.23s/it]


                   all        641       3701      0.816      0.565      0.723       0.52
            Pedestrian        182        332      0.735       0.56      0.691      0.449
         Traffic_Signs        168        230      0.818       0.47      0.671      0.448
               Vehicle        528       1377      0.891      0.828      0.898      0.769
         traffic_light        338       1762      0.819      0.404       0.63      0.413
Speed: 1.8ms preprocess, 1.6ms inference, 0.0ms loss, 0.7ms postprocess per image
Saving runs/detect/val3/predictions.json...
Results saved to runs/detect/val3


# Best Checkpoint+ Taking the Optimal value from the F1-Confidence curve for Optimal Confidence
- Optimal Confidence YOLOv10n - 0.264
- IoU threshold : 0.5
- Results save d at: runs/detect/val4

In [1]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.264, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.5, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLOv10n summary (fused): 285 layers, 2,695,976 parameters, 0 gradients, 8.2 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:06<00:00,  1.16s/it]


                   all        641       3701      0.812      0.562      0.719      0.518
            Pedestrian        182        332      0.717      0.557      0.684      0.446
         Traffic_Signs        168        230      0.817      0.465      0.667      0.446
               Vehicle        528       1377      0.895      0.825      0.897      0.768
         traffic_light        338       1762      0.819        0.4      0.629      0.411
Speed: 1.9ms preprocess, 1.6ms inference, 0.0ms loss, 0.7ms postprocess per image
Saving runs/detect/val4/predictions.json...
Results saved to runs/detect/val4


# Best Checkpoint (99) + Optimal Confidence + IoU = 0.6
- Optimal Confidence YOLOv10n - 0.264
- IoU threshold : 0.6
- Results save d at: runs/detect/val5

In [1]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.264, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLOv10n summary (fused): 285 layers, 2,695,976 parameters, 0 gradients, 8.2 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:06<00:00,  1.12s/it]


                   all        641       3701      0.812      0.562      0.719      0.518
            Pedestrian        182        332      0.717      0.557      0.684      0.446
         Traffic_Signs        168        230      0.817      0.465      0.667      0.446
               Vehicle        528       1377      0.895      0.825      0.897      0.768
         traffic_light        338       1762      0.819        0.4      0.629      0.411
Speed: 1.9ms preprocess, 1.7ms inference, 0.0ms loss, 0.7ms postprocess per image
Saving runs/detect/val5/predictions.json...
Results saved to runs/detect/val5


# Seperate evaluation on the training set

In [1]:
from ultralytics import YOLO

# Load your model
model = YOLO('/home/sur06423/project/Version_5_exp/YOLO_v10/v10_n/train/weights/best.pt')  # replace with your model path

# Validate on custom dataset
results = model.val(data='/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml',
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the optimal confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="train", 
)  # Ensure your YAML points to your desired dataset

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLOv10n summary (fused): 285 layers, 2,695,976 parameters, 0 gradients, 8.2 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/train/labels.cache... 4485 images, 430 backgrounds, 0 corrupt: 100%|██████████| 4485/4485 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 36/36 [00:20<00:00,  1.77it/s]


                   all       4485      26788      0.833      0.603      0.752      0.555
            Pedestrian       1229       2227      0.785       0.63       0.75      0.517
         Traffic_Signs       1093       1459      0.819       0.52      0.702       0.47
               Vehicle       3636      10031      0.894      0.857      0.915       0.79
         traffic_light       2373      13071      0.835      0.406      0.641      0.444
Speed: 0.4ms preprocess, 1.3ms inference, 0.0ms loss, 0.1ms postprocess per image
Saving runs/detect/val6/predictions.json...
Results saved to runs/detect/val6


# This notebook is complete.
- All the cell are executed and saved for later use.
- Results are saved in the corresponding directories.